### PR TITLE
Lecture7

### PR DESCRIPTION
**Pull request by VictorLVA**

### CODE REVIEW

File: Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        WebElement activeCategory = driver.findElement(By.className("catalog-navigation-list__aside-item_active"));
+        List<WebElement> activeCategoryChapters = activeCategory.findElements(By.className("catalog-navigation-list__dropdown-item"));
+        int activeCategoryChaptersIndex = (int) (Math.random() * activeCategoryChapters.size());
+        System.out.println("Chapter: " + activeCategoryChapters.get(activeCategoryChaptersIndex)
+                                                               .findElement(By.className("catalog-navigation-list__dropdown-title"))
+                                                               .getText());
+        activeCategoryChapters.get(activeCategoryChaptersIndex).click();
+    }
+
+    public static void openOnlinerRandomProductWithOffers(WebDriver driver, Wait<WebDriver> waiter) {
```
Comment (acherkas) [inline Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java]: why not use like variable for all class Helper?

File: Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    attemps -= 1;
+                    System.out.println("No offers for products with the selected chapter. Trying another chapter (" + attemps + " attemps left)");
+                    openOnlinerFullCatalog(waiter);
+                    openOnlinerRandomCatalogChapter(driver);
+                }
+            }
+        }
+    }
+
+    public static void openOnlinerCart(Wait<WebDriver> waiter) {
```
Comment (acherkas) [inline Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java]: the same about waiter

File: Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerProductsManipulation.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import static Helper.OnlinerUtils.OnlinerNavigation.openOnlinerFullCatalog;
+import static Helper.OnlinerUtils.OnlinerNavigation.openOnlinerRandomCatalogChapter;
+import static Helper.OnlinerUtils.OnlinerNavigation.openOnlinerRandomProductWithOffers;
+
+public class OnlinerProductsManipulation {
+
+    private OnlinerProductsManipulation() {
+    }
+
+    public static void addProductToCartWithRandomOffer(WebDriver driver, Wait<WebDriver> waiter) {
```
Comment (acherkas) [inline Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerProductsManipulation.java]: waiter?

File: Lecture7/Testing_Task1/Helper/WebDriverUtils/ChromeDriverSettings.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package Helper.WebDriverUtils;
+
+class ChromeDriverSettings {
```
Comment (acherkas) [inline Lecture7/Testing_Task1/Helper/WebDriverUtils/ChromeDriverSettings.java]: Didn't catch the logic of this class
You can use like fields variables in this class, but if you are returning constants so use it like constants

File: Lecture7/Testing_Task1/Helper/WebDriverUtils/WebDriverUtils.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import static Helper.WebDriverUtils.ChromeDriverSettings.getDriverName;
+import static Helper.WebDriverUtils.ChromeDriverSettings.getLocation;
+
+public class WebDriverUtils {
+
+    private WebDriverUtils() {
+    }
+
+    public static WebDriver useChromeWebDriver() {
+        System.setProperty(getDriverName(), getLocation());
```
Comment (acherkas) [inline Lecture7/Testing_Task1/Helper/WebDriverUtils/WebDriverUtils.java]: constants or set properties in @BeforeClass . Test

File: Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerProductsManipulation.java (hunk 3 above)
Comment (VictorLVA) [reply Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerProductsManipulation.java]: fixed

File: Lecture7/Testing_Task1/Helper/WebDriverUtils/ChromeDriverSettings.java (hunk 4 above)
Comment (VictorLVA) [reply Lecture7/Testing_Task1/Helper/WebDriverUtils/ChromeDriverSettings.java]: fixed

File: Lecture7/Testing_Task1/Helper/WebDriverUtils/WebDriverUtils.java (hunk 5 above)
Comment (VictorLVA) [reply Lecture7/Testing_Task1/Helper/WebDriverUtils/WebDriverUtils.java]: fixed

File: Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java (hunk 1 above)
Comment (VictorLVA) [reply Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java]: fixed

File: Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java (hunk 2 above)
Comment (VictorLVA) [reply Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java]: fixed

**Pull request merged**